eng/native/configurecompiler.cmake

```diff
@@ -105,6 +105,10 @@ elseif (CLR_CMAKE_HOST_UNIX)
   # Use uppercase CMAKE_BUILD_TYPE for the string comparisons below
   string(TOUPPER ${CMAKE_BUILD_TYPE} UPPERCASE_CMAKE_BUILD_TYPE)
 
+  if (NOT CLR_CMAKE_HOST_ARCH_S390X AND NOT CLR_CMAKE_HOST_OSX AND NOT CLR_CMAKE_HOST_FREEBSD)
```

Comment:
```suggestion
  if (NOT CLR_CMAKE_HOST_ARCH_S390X AND NOT CLR_CMAKE_HOST_OSX AND NOT CLR_CMAKE_HOST_FREEBSD AND NOT CMAKE_C_COMPILER_ID MATCHES "GNU")
```

Reply:
and I still think https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/eng/native/configuretools.cmake#L67 would be a better place for this to keep linker configs in one place (unless there is a reason not to).

Reply:
So you are saying that instead of detecting the linker there for Linux, we would set this option at that place? We add a plenty of linker flags in the eng/native/configurecompiler.cmake already, so it seemed to make sense to have it there.

Reply:
I was actually suggesting to move that condition after line 67 of configuretools.cmake (without deleting anything existing there), because that is the file which is included in all subsets including libraries. configurecompiler is only for coreclr and hosts. Unfortunately, it is not clear as we haven't finished the refactoring properly from my point of view but I paused it to avoid merge conflicts at the time.

I think you would agree that it is good idea to use same linker (whichever is selected) for all components of a given build rather than mixing.

Reply:
Yes, I agree with using the same linker for everything. As for moving the setting, it seemed strange that we would set the lld using the option and then right away run the linker detection and find that it is lld.

Reply:
So I was thinking I might have misunderstood what you've meant.

Reply:
We could be more explicit, e.g.

```cmake
if (NOT CLR_CMAKE_HOST_WIN32)
   if (NOT CLR_CMAKE_HOST_ARCH_S390X AND NOT CLR_CMAKE_HOST_OSX AND NOT CLR_CMAKE_HOST_FREEBSD)
       add_linker_flag(-fuse-ld=lld)
       set(LD_GNU 1) # or maybe LD_LLVM and update the usages with `LD_GNU OR LD_LLVM` to be more explicit / future proof?)
  else()
    # detect linker
    ... existing code ..
  endif()
endif()
```